modules/mobile/app/models/mobile/v0/adapters/appointment_requests.rb

```diff
@@ -0,0 +1,239 @@
+# frozen_string_literal: true
+
+require 'sentry_logging'
+
+module Mobile
+  module V0
+    module Adapters
+      class AppointmentRequests
+        # accepts an array of appointment requests
+        # returns a list of appointments, filtering out any that are not SUBMITTED or CANCELLED
+        def parse(requests)
+          va_appointments = []
+          cc_appointments = []
+
+          requests.each do |request|
+            status = status(request)
+            next unless status.in?(%w[CANCELLED SUBMITTED])
+
+            if request.cc_appointment_request
+              cc_appointments << build_appointment_model(request, CC)
+            else
+              va_appointments << build_appointment_model(request, VA)
+            end
+          end
+
+          [va_appointments, cc_appointments]
+        end
+
+        private
+
+        # rubocop:disable Metrics/MethodLength
+        def build_appointment_model(request, klass)
+          start_date = localized_start_date(request)
+
+          Mobile::V0::Appointment.new(
+            id: request[:appointment_request_id],
+            appointment_type: appointment_type(request, klass),
+            cancel_id: nil,
+            comment: nil,
+            facility_id: klass.facility_id(request),
+            sta6aid: nil,
+            healthcare_provider: klass.provider_name(request),
+            healthcare_service: klass.practice_name(request),
+            location: klass.location(request),
+            minutes_duration: nil,
+            phone_only: phone_only?(request),
+            start_date_local: start_date,
+            start_date_utc: start_date.utc,
+            status: status(request),
+            status_detail: status_detail(request),
+            time_zone: time_zone(request),
+            vetext_id: nil,
+            reason: request[:purpose_of_visit],
+            is_covid_vaccine: nil,
+            is_pending: true,
+            proposed_times: proposed_times(request),
+            type_of_care: request[:appointment_type],
+            patient_phone_number: request[:phone_number],
+            patient_email: request[:email],
+            best_time_to_call: request[:best_timeto_call],
+            friendly_location_name: request[:friendly_location_name]
```

Comment:
We want both the facility name and the friendly name. I didn't add this to the location data because that's used in other places. It's also not embedded in `location` in the source data.